include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -11,125 +11,110 @@
-    {
-      return _mm256_loadu_si256((__m256i *)p);
-    }
-  }
+    using r_t = as_wide_t<std::remove_cvref_t<T>>;
 
-  template<real_scalar_value T, typename N, std::size_t A>
-  EVE_FORCEINLINE auto
-  load(eve::as_<wide<T, N>> const &tgt, eve::x86_256_ const &mode, aligned_ptr<T const, A> p) noexcept
-  {
-    if constexpr( A >= 32 )
+    // If the ignore/keep is complete we can jump over if_else
+    if constexpr( !C::is_complete )
     {
-      if constexpr( std::is_same_v<T, double> )
+      r_t that;
+      auto mask = cond.mask( as_<r_t>{} ).bits();
+
+      constexpr auto cat = categorize<r_t>();
+
+            if constexpr( cat == category::float64x4 )  that = _mm256_maskload_pd (ptr,mask);
+      else  if constexpr( cat == category::float64x2 )  that = _mm_maskload_pd    (ptr,mask);
+      else  if constexpr( cat == category::float32x8 )  that = _mm256_maskload_ps (ptr,mask);
+      else  if constexpr( cat == category::float32x4 )  that = _mm_maskload_ps    (ptr,mask);
+      else  if constexpr( current_api >= avx2 )
       {
-        return _mm256_load_pd(p.get());
+        constexpr auto m64x4 = match(cat, category::int64x4, category::uint64x4);
+        constexpr auto m64x2 = match(cat, category::int64x2, category::uint64x2);
+        constexpr auto m32x8 = match(cat, category::int32x8, category::uint32x8);
+        constexpr auto m32x4 = match(cat, category::int32x4, category::uint32x4);
+
+              if constexpr( m64x4 ) that = _mm256_maskload_epi64 (ptr,mask);
+        else  if constexpr( m64x2 ) that = _mm_maskload_epi64    (ptr,mask);
+        else  if constexpr( m32x8 ) that = _mm256_maskload_epi32 (ptr,mask);
+        else  if constexpr( m32x4 ) that = _mm_maskload_epi32    (ptr,mask);
+        else                        return load_(EVE_RETARGET(cpu_), cond, ptr);
       }
-      else if constexpr( std::is_same_v<T, float> )
+      else
       {
-        return _mm256_load_ps(p.get());
+        return  load_(EVE_RETARGET(cpu_), cond, ptr);
       }
-      else if constexpr( std::is_integral_v<T> )
+
+      if constexpr( C::has_alternative )
       {
-        return _mm256_load_si256((__m256i *)p.get());
+        constexpr auto blender = [](auto a, auto b, auto m)
+        {
+          auto mm = typename r_t::storage_type(m.storage());
+                if constexpr( cat == category::float64x4 )  return _mm256_blendv_pd(a,b,mm);
+          else  if constexpr( cat == category::float32x8 )  return _mm256_blendv_ps(a,b,mm);
+          else  if constexpr( cat == category::float64x2 )  return _mm_blendv_pd(a,b,mm);
+          else  if constexpr( cat == category::float32x4 )  return _mm_blendv_ps(a,b,mm);
+          else  if constexpr( sizeof(a) == 32)              return _mm256_blendv_epi8(a,b,mm);
+          else  if constexpr( sizeof(a) == 16)              return _mm_blendv_epi8(a,b,mm);
+        };
+
+        r_t alt(cond.alternative);
+        that = blender(alt,that,mask);
```

Comment:
Why not just call `if_else`?  Trying to save on compile time?

Reply:
if_else is in core an I want to minimize the internals details dependency on outside function.
It also saves compiles time as we don't need the full power of if_else special cases handling.

Reply:
Ok. You could put it in some common place though. But I don't care much.